### PR TITLE
Added Apollo Tracing support for `@defer`.

### LINKED ISSUE #1600: Defer: ftv1 support
Followup to 1599 where i disabled sending traces for defered queries, we want to have a look at what needs to be done so we can support it

### PR DESCRIPTION
**Pull request by BrynCooke**
You can now view traces in Apollo Studio as normal.

Fixes #1600

The main file to review is: apollo-router/src/plugins/telemetry/mod.rs
I've added a bunch of testing, but have not enhanced the jaeger test as I think we want to use tracing test for this.

### CODE REVIEW

File: xtask/src/commands/compliance.rs [hunk 1]
```diff
@@ -12,7 +12,11 @@ use xtask::*;
 static LICENSES_HTML_PATH: &str = "licenses.html";
 
 #[derive(Debug, StructOpt)]
-pub struct Compliance {}
+pub struct Compliance {
+    /// Do not run the compliance test
+    #[structopt(long)]
+    skip_compliance: bool,
```
Comment (BrynCooke) [inline xtask/src/commands/compliance.rs:18]: Added this option to allow us to run tests until we have upgraded the project for new cargo about.

File: apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap [hunk 2]
```diff
@@ -399,7 +394,7 @@ expression: get_spans()
                   "record": {
                     "entries": [
                       [
-                        "apollo_private.path",
+                        "path",
```
Comment (BrynCooke) [inline apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap:397]: Yes path is not something that appears in our traces.

**Review by bnjjj — APPROVED** [review]

File: apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap (hunk 2 above)
Comment (bnjjj) [reply apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap:397]: Should we namespace it to be more otel specs compliant ?

File: apollo-router/src/plugins/telemetry/tracing/apollo.rs [hunk 3]
```diff
@@ -28,7 +28,6 @@ impl TracingConfigurator for Config {
 
                 let exporter = apollo_telemetry::Exporter::builder()
                     .expose_trace_id_config(expose_trace_id.clone())
-                    .trace_config(trace_config.clone())
```
Comment (Geal) [inline apollo-router/src/plugins/telemetry/tracing/apollo.rs:31]: why is this removed?

Comment (BrynCooke) [reply apollo-router/src/plugins/telemetry/tracing/apollo.rs:31]: It was unused. We stashed it in a struct, but never accessed it. Not sure why the compiler didn't pick it up.

File: apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap (hunk 2 above)
Comment (BrynCooke) [reply apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap:397]: Yes, Will make the change

Comment (BrynCooke) [reply apollo-router/tests/snapshots/integration_tests__traced_basic_composition.snap]: Fixed in [38b5983](https://github.com/apollographql/router/pull/2190/commits/38b59835e2b322ecf9ac74af0cdcf010eb32a08c)

File: apollo-router/src/query_planner/execution.rs [hunk 4]
```diff
@@ -200,66 +213,63 @@ impl PlanNode {
...
-                        value.deep_merge(v);
-                        errors.extend(err.into_iter());
-                        subselection = primary_subselection.clone();
 
-                        let _ = primary_sender.send(value.clone());
-                    } else {
-                        let _guard = span.enter();
+                        tokio::task::spawn(async move {
+                            join_all(futures).await;
+                        });
```
Comment (garypen) [inline apollo-router/src/query_planner/execution.rs:241]: Previously this was `in_current_span`, why was that removed? (I imagine because we were "double counting" with `instrument`, but I want to make sure I understand.)

**Review by garypen — APPROVED** [review]
Just one question. The `span` stuff is hairy, but I think the changes might have cleaned things up there.

File: apollo-router/src/query_planner/execution.rs (hunk 4 above)
Comment (garypen) [reply apollo-router/src/query_planner/execution.rs:241]: Discussed with @BrynCooke . The `in_current_span()` was redundant.

Comment (BrynCooke) [reply apollo-router/src/query_planner/execution.rs:241]: @garypen and I discussed this offline. There are several interesting points.
* A span is parented upon create - not first entry.
* A span cannot close before all children are closed. If this is desired then `follow-on` must be used.

We had code that called:
```rust
.instrument(span) // The new span
.in_current_span() // The current span at invocation
```
The future gets wrapped by a 2 wrapper that set the span.
When the future is polled the outer wrapper set the span, but then it is immediately set again by the inner wrapper. So there is no point in the outer wrapper.
